docs/side_quests/working_with_files.md

```diff
@@ -0,0 +1,1016 @@
+//             type: type,
+//             readNum: readNum,
+//         ],
+//         myFile
+//     ]
+// }
+.view()
+```
+
+Run the workflow:
+
+```bash
+nextflow run main.nf
+```
+
+The output will show the paired files grouped together:
+
+```console title="Channel.fromFilePairs Output"
+ N E X T F L O W   ~  version 24.10.4
+
+Launching `main.nf` [chaotic_cuvier] DSL2 - revision: 472265a440
+
+[sampleA_rep1_normal_R, [/workspaces/training/side-quests/working_with_files/data/sampleA_rep1_normal_R1_001.fastq.gz, /workspaces/training/side-quests/working_with_files/data/sampleA_rep1_normal_R2_001.fastq.gz]]
+```
+
+Note the difference in data structure. Rather than being a list of results, we have a single result in the format `id, [ fastq1, fastq2 ]`. Nextflow has done the hard work of extracting the sample name by examining the shared prefix and using it as a sample id.
+
+### 4.2. Extract metadata from file pairs
+
+We still need the metadata. Our `map` operation from before won't work because it doesn't match the data structure, but we can modify it to work. We already have access to the sample name in the `id` variable, so we can use that to extract the metadata without grabbing the `simpleName` from the file object like before.
+
+_Before_:
+
+```groovy title="main.nf" linenums="3"
+ch_fastq = Channel.fromFilePairs('data/sampleA_rep1_normal_R{1,2}_001.fastq.gz')
+// ch_fastq.map { myFile ->
+//     def (sample, replicate, type, readNum) = myFile.simpleName.tokenize('_')
+//     [
+//         [
+//             id: sample,
+//             replicate: replicate.replace('rep', ''),
+//             type: type,
+//             readNum: readNum,
+//         ],
+//         myFile
+//     ]
+// }
+.view()
+```
+
+_After_:
+
+```groovy title="main.nf" linenums="3"
+ch_fastq = Channel.fromFilePairs('data/sampleA_rep1_normal_R{1,2}_001.fastq.gz')
+ch_fastq.map { id, fastqs ->
+    def (sample, replicate, type, readNum) = id.tokenize('_')
+    [
+        [
+            id: sample,
+            replicate: replicate.replace('rep', ''),
+            type: type,
+            readNum: readNum,
+        ],
+        fastqs
+    ]
+}
+.view()
+```
+
+```bash
+nextflow run main.nf
+```
+
+```console title="File Pairs Output parsed"
+
+ N E X T F L O W   ~  version 24.10.5
+
+Launching `main.nf` [prickly_stonebraker] DSL2 - revision: f62ab10a3f
+
+[[id:sampleA, replicate:1, type:normal, readNum:R], [/Users/adam.talbot/training/side-quests/working_with_files/data/sampleA_rep1_normal_R1_001.fastq.gz, /Users/adam.talbot/training/side-quests/working_with_files/data/sampleA_rep1_normal_R2_001.fastq.gz]]
+```
+
+Well done! We have grabbed the metadata from the filenames and used them as values in the tuple.
+
+### Takeaway
+
+- `Channel.fromFilePairs()` automatically finds and pairs related files
+- The closure defines how to extract sample IDs from filenames
+- This simplifies handling paired-end reads in your pipeline
```

Comment:
could you link to the docs?
There are some nice tricks there, especially the part where you can deal with paired-end and single end files at the same time.